app/controllers/general.php

```diff
@@ -289,19 +291,66 @@
         ->noContent();
 }, ['request', 'response']);
 
-App::error(function ($error, $utopia, $request, $response, $layout, $project) {
+App::error(function ($error, $utopia, $request, $response, $layout, $project, $logger, $user, $loggerBreadcrumb) {
     /** @var Exception $error */
     /** @var Utopia\App $utopia */
     /** @var Utopia\Swoole\Request $request */
     /** @var Appwrite\Utopia\Response $response */
     /** @var Utopia\View $layout */
     /** @var Appwrite\Database\Document $project */
+    /** @var Utopia\Logger\Logger $logger */
+    /** @var Appwrite\Database\Document $user */
+
+    $version = App::getEnv('_APP_VERSION', 'UNKNOWN');
+    $route = $utopia->match($request);
+
+    if($error->getCode() >= 500 || $error->getCode() === 0) {
+        $log = new Utopia\Logger\Log();
+
+        if(!$user->isEmpty()) {
+            $log->setUser(new User($user->getId()));
+        }
+
+        $log->setNamespace("http");
+        $log->setServer(App::getEnv("_APP_LOGGING_SERVERNAME", "selfhosted-001"));
+        $log->setVersion($version);
+        $log->setType(Log::TYPE_ERROR);
+        $log->setMessage($error->getMessage());
+
+        $log->setTags([
```

Comment:
This is more related to the library implementation. I would like us to avoid signatures that expect assoc array as much as possible. This usually creates a loose interface that is not strictly typed.

This implementation should be closer to our usual pattern:
```php
$log
    ->addTag('string1', 'stringX')
    ->addTag('string2', 'stringY')
;
```

Reply:
Added to utopia-php/logger: https://github.com/utopia-php/logger/commit/5d8cc1bbbfc738ae28ede0d438a1d5318011cf6a

Applied this to: `tags` named array, `breadcrumbs` array, `extra` named array.